src/app/auth/sign-up/sign-up.component.html

```diff
@@ -0,0 +1,42 @@
+<form [formGroup]="signUpForm"
+  (ngSubmit)="onSubmit()">
+  <mat-form-field>
+    <mat-label>Username</mat-label>
+    <input matInput
+      formControlName="displayName"
+      name="displayName"
+      type="text"
+      required />
+    <mat-error *ngIf="hasError('email', 'required')">Please enter your email</mat-error>
+  </mat-form-field>
+  <mat-form-field>
+    <mat-label>Email</mat-label>
+    <input matInput
+      formControlName="email"
+      name="email"
+      type="text"
+      required />
+    <mat-error *ngIf="hasError('email', 'required')">Please enter your email</mat-error>
+  </mat-form-field>
+  <mat-form-field>
+    <mat-label>Password</mat-label>
+    <input matInput
+      formControlName="password"
+      name="password"
+      [type]="showPassword ? 'text' : 'password'" />
+    <mat-icon matSuffix
+      (click)="showPassword = !showPassword">
+      {{ showPassword ? 'visibility_off' : 'visibility' }}
+    </mat-icon>
+    <mat-error *ngIf="hasError('password', 'required')">Please enter your password</mat-error>
```

Comment:
you have validator about length, but don't process this error

Reply:
Fixed